RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+Once the destination node receives and decrypts the data message, that node is then able to establish a P2P communication channel with the source node for any further communication.
+
+Propagation of completely private discovery request, hidden as an encrypted data message, can be performed as a broadcast through the entire network using the Gossip protocol.
+Propagation of public discovery requests can be performed using more efficient directed propagation using the Kademlia protocol.
+As encrypted message with visible destinations tend to not be of interest to the rest of the network, directed propagation using the Kademlia protocol to forward these messages to the correct parties are preferred.
+
+This same encryption technique can be used to send encrypted messages to a single node or a group of nodes, where the group of nodes have shared identification keys.
+A Validation Committee is an example of a group of CNs that have shared identification keys for the committee.
+The shared identification keys ensure that all members of that committee are able to receive and decrypt data messages that were sent to the committee.
+
+#### Maintaining connections with peers
+
+CCs and CNs establish and maintain connections with peers differently.
+CCs only create a few short-lived ad hoc channels and CNs create and maintain long-lived channels with a number of peers.
+
+If a CC is unaware of a destination CNs or CCs online communication address then the address first needs to be obtained using a discovery request.
+When a CC already know the communication address of the CC or CN that he wants to communicate with, then a direct P2P channel can be established between the two peers for the duration of the communication task.
+The communication channel can then be closed as soon as the communication task has been completed.
+
+CNs consisting of VNs and BNs typically attempt to maintain communication channels with a large number of peers.
+The distribution of peers (VNs vs BNs) that a single CN keeps communication channels open with can change depending on the type of node.
+A CN that is also a BN should maintain more peer connections with other BNs, but should also have some connections with other VNs.
+
+Having some connections with VNs are important as BNs have derived node ids and not registered node ids such as VNs, making it possible for the CN to be separated from the main network and become victim to an eclipse attack.
+Having some connections with VNs will make it more difficult to separate the CN from the network and will ensure successful propagation of transactions and completed blocks from that CN. 
+
+A CN that is also a VN should maintain more peer connections with other VNs, but also have some connections with BNs.
+CNs that are part of Validator Node committees should attempt to maintain permanent connections with the other members of the committee to ensure that quick consensus can be achieved.
```

Comment:
Perhaps add here that the type of p2p channel is open for debate between the nodes. Do they want tor, i2p or for example something faster like ipv4?

Reply:
From the proposed network address type it is up to the CC or CN to decide what they want to use as an online address. I think that IPv4, IPv6, URL, Tor(Base32) and I2P(Base32) addresses should be allowed.

Reply:
The owner of a wallet may want the fact that they are transacting to be as private as possible. We may rate the different choices so users can select the appropriate mechanisms.

Reply:
I wasn't sure how to rate them, so I added the following small paragraph: "A Tor or I2P address can be used when anonymity is important for a CC or CN. The IPv4, IPv6 and URL address types do not provide any privacy features but do provide increased bandwidth. "

Reply:
Good for me, thanks. Maybe consider
"_A Tor or I2P address **SHOULD** be used when anonymity is important for a CC or CN._"

Reply:
Perhaps we should add a list of addresses type thing, as a CC or CN might not have a Tor or I2P address.

Reply:
I think it is clear from the "online communication address type" that there are 5 options:  IPv4, IPv6, URL, Tor (Base32) or I2P (Base32) address

Reply:
What I mean is that we should allow users to have more than one address. 

The example I have is that a VN might want to give out say a tor address, but he might want to give a IPv4 address for his VN peers.

Reply:
With the current structure a CN could give out different addresses to other CNs and CCs, but every CC or CN will only have one peer address with one network address in their routing table. Having a list of network addresses for each peer address could be a future improvement.